api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto

```diff
@@ -259,6 +289,9 @@ message RedisProxy {
   // configuring the backing clusters.
   PrefixRoutes prefix_routes = 5;
 
+  // We can add validation to allow only on router config
+  HashSlotRoutes hash_slot_routes = 10;
```

Comment:
Technically putting both of these inside a oneof is wire compatible, so I would recommend we just go ahead and do that. In general we don't allow this to avoid churn in language compiles, but this isn't a major enough configuration point that I think it's probably worth it to leave this clean.